### PR TITLE
feat: display fiat in send form

### LINKED ISSUE #3191: Display fiat value below crypto amount in send form
We should show the fiat conversion value below the amount entered in STX or BTC here:

image

It currently looks like this:

Screenshot 2023-02-18 at 13 46 43

Note that https://github.com/hirosystems/stacks-wallet-web/issues/3189 is for adding the option to swap whereas this issue is specifically for displaying the fiat value only (and not being able to swap the values). So this issue should come first.

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/4315940170).

Closes https://github.com/hirosystems/stacks-wallet-web/issues/3191

### CODE REVIEW

**Comment by alter-eggo** [comment]
@kyranjamie @fbwoolf could you please check if I go in right direction from the code side, assuming that the next task is to implement switching fiat/asset amount

**Comment by alter-eggo** [comment]
now fiat calculation works only for btc, if code is ok, I will implement for stx as well

**Comment by alter-eggo** [comment]
it seems that it will be necessary to create one more `<AmountField />` for fiat input mode and separate component for `switchableAmount`, where asset amount will be calculated for given fiat amount

and just switch these two blocks (I'll do it in separate pr)

File: src/app/pages/send/send-crypto-asset-form/components/send-fiat-value.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { Text } from '@stacks/ui';
+import { useField } from 'formik';
+
+import { MarketData } from '@shared/models/market.model';
+import { Money, createMoneyFromDecimal } from '@shared/models/money.model';
+
+import { baseCurrencyAmountInQuote } from '@app/common/money/calculate-money';
+import { i18nFormatCurrency } from '@app/common/money/format-money';
+
+interface SenfFiatInputProps {
```
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/components/send-fiat-value.tsx]: Send*

File: src/app/pages/send/send-crypto-asset-form/components/send-fiat-value.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export function SendFiatValue({ marketData, assetSymbol = '' }: SenfFiatInputProps) {
+  const [field] = useField('amount');
+  const [assetValue, setAssetValue] = useState<Money>(createMoneyFromDecimal(0, assetSymbol));
+
+  useEffect(() => {
+    let amount = Number(field.value);
+
+    if (isNaN(amount)) {
+      amount = 0;
+    }
```
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/components/send-fiat-value.tsx]: Could also be other things right, undefined etc potentially?

**Review by kyranjamie — APPROVED** [review]
LGTM, nice approach of passing in JSX 👍🏼

File: src/app/pages/send/send-crypto-asset-form/components/send-fiat-value.tsx (hunk 2 above)
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/components/send-fiat-value.tsx]: hm, I think not, but added condition to be sure

**Comment by mica000** [comment]
Screenshot 2023-02-28 at 18 15 22
@kyranjamie @markmhx 

Should this PR also remove the ability to copy a BTC address when switching accounts or this is being done in another place?

**Comment by markmhendrickson** [comment]
@mica000 let's handle that removal elsewhere. Mind creating an issue with design guidance on just how we should remove?

**Comment by alter-eggo** [comment]
sorry, I completely messed up with git, I am trying to fix it now

**Comment by alter-eggo** [comment]
git flow is fixed, but tests still failing. fixing them now

**Comment by alter-eggo** [comment]
@kyranjamie @fbwoolf can you please help here, please? 
playwrite test `that the address cannot be same as sender` is failing because I added autofocus on send form input field. so when user selects same address and clicks on preview btn, `Address provided is not valid` error is shown for a second and then changes to the right one `Cannot send to yourself`. 

actually it is the same problem in dev branch now, if you go to stx send form, click on input field, and then choose current address you will see the above weird problem.

**Comment by markmhendrickson** [comment]
image

Looking good! Though it appears we need a bit more top and bottom margin compared to the design spec, the text could be bolder, and we should place "USD" at the end:

image

[See Figma](https://www.figma.com/file/OSQocEEJnfcMkniEl8u3U8/%231483---Initial-Bitcoin-support?node-id=1065%3A58140&t=9q5d7PIsMeYRboBe-1)

**Comment by mica000** [comment]
> @mica000 let's handle that removal elsewhere. Mind creating an issue with design guidance on just how we should remove?

Yes, new issue here:
https://github.com/hirosystems/stacks-wallet-web/issues/3327

**Comment by alter-eggo** [comment]
@kyranjamie @fbwoolf here is the issue I mentioned, when you click preview btn, first error is `Address is not valid` and then it changes to `Cannot send to you`, that is why tests are failing

https://user-images.githubusercontent.com/46521087/222201855-5a1892c5-da27-40ff-a38a-24d9be17d0ea.mov

**Comment by alter-eggo** [comment]
weird, now playwright tests passed, but nevertheless not sure we can merge it as is

**Comment by markmhendrickson** [comment]
@alter-eggo need another code review here or should we try merging and see what happens?

**Comment by alter-eggo** [comment]
@markmhx not sure here, from the code side it's ok, but I reran playwright tests and they are still failing. 

in order not to block this pr we can either:
- remove autofocus
- remove autofocus in test mode `autofocus={!IS_TEST_ENV}`, 
- comment that specific test

and create separate task to fix this problem.

I think removing autofocus in test mode might be ok as temporary solution, if I'm not mistaken @kyranjamie you advised this as well?

**Comment by kyranjamie** [comment]
if disabling autofocus helps for now, all good

**Pull request merged**